formatter.nix

```diff
@@ -0,0 +1,42 @@
+{ pkgs, inputs, ... }:
+inputs.treefmt-nix.lib.mkWrapper pkgs {
+  projectRootFile = ".git/config";
+
+  programs = {
+    nixfmt.enable = true;
+    deadnix.enable = true;
+    gofumpt.enable = true;
+    prettier.enable = true;
+    statix.enable = true;
+  };
+
+  settings = {
+    global.excludes = [
+      "LICENSE.txt"
+      # unsupported extensions
+      "*.{gif,png,svg,tape,mts,lock,mod,sum,toml,env,envrc,gitignore}"
+    ];
+
+    formatter = {
+      deadnix = {
+        priority = 1;
+      };
+
+      statix = {
+        priority = 2;
+      };
+
+      nixfmt = {
+        priority = 3;
+      };
+
+      prettier = {
+        options = [
+          "--tab-width"
+          "2"
+        ];
+        includes = [ "*.{css,html,js,json,jsx,md,mdx,scss,ts,yaml}" ];
+      };
+    };
+  };
+}
```

Comment:
Uses [treefmt](https://treefmt.com/) to format all the things. You can format the whole repo with `nix fmt`.